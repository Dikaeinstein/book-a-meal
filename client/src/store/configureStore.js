import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import { createLogger } from 'redux-logger';
import throttle from 'lodash.throttle';
import { loadState, saveState } from '../helpers/localStorage';
import initialState from '../reducers/initialState';
import rootReducer from '../reducers';

/**
 * Return redux middleware
 */
const getMiddleware = () => {
  const logger = createLogger();

  const middleware =
    process.env.NODE_ENV !== 'production'
    && process.env.NODE_ENV !== 'test'
      ? applyMiddleware(thunk, logger)
      : applyMiddleware(thunk);

  return middleware;
};

/**
 * Configure the redux store
 *
 * @returns {object} Redux store
 */
const configureStore = () => {
  const persistedState = loadState();

  const state = persistedState
    ? { ...initialState, ...persistedState }
    : initialState;

  const store = createStore(
    rootReducer,
    state,
    compose(
      getMiddleware(),
      window.devToolsExtension ? window.devToolsExtension() : empty => empty,
    ),
  );

  // Throttle the saveState function to be called once every second
  store.subscribe(throttle(() => {
    saveState({ orders: store.getState().orders });
  }, 1000));

  return store;
};

export default configureStore;
