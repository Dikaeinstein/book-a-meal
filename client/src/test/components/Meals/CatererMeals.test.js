import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import ConnectedCatererMeals, { CatererMeals }
  from '../../../components/Meals/CatererMeals';

describe('<CatererMeals />', () => {
  const meals = [
    { id: 1, name: 'test meal', price: 4000 },
    { id: 2, name: 'test meal2', price: 4000 },
  ];
  const props = {
    isFetching: false,
    handleMealUpdate: jest.fn(),
    meals,
  };
  it('should render with correct props from redux', () => {
    const mockStore = configureStore();
    const store = mockStore({
      meals: {
        isFetching: false,
        byId: { 1: meals[0], 2: meals[1] },
        allIds: [1, 2],
      },
    });
    const wrapper = shallow(<ConnectedCatererMeals
      store={store}
      handleMealUpdate={jest.fn()}
    />);
    expect(wrapper.props().isFetching).toBeFalsy();
    expect(wrapper.props().meals.length).toEqual(2);
  });
  it('should match snapshot', () => {
    let tree = shallow(<CatererMeals {...props} />);
    expect(tree).toMatchSnapshot();
    tree = shallow(<CatererMeals {...{ ...props, isFetching: true }} />);
    expect(tree).toMatchSnapshot();
  });
});
