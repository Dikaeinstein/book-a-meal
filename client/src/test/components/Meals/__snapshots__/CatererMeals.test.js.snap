// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CatererMeals /> should match snapshot 1`] = `
<section
  className="bg-light"
>
  <h4
    className="text-center"
  >
    Meals
  </h4>
  <CatererMealList
    handleMealUpdate={[MockFunction]}
    meals={
      Array [
        Object {
          "id": 1,
          "name": "test meal",
          "price": 4000,
        },
        Object {
          "id": 2,
          "name": "test meal2",
          "price": 4000,
        },
      ]
    }
  />
</section>
`;

exports[`<CatererMeals /> should match snapshot 2`] = `
<div
  className="loader-container"
>
  <Loading
    text="fetching meals..."
  >
    <Preloader
      active={true}
      color="blue"
      flashing={true}
      size="medium"
    />
  </Loading>
</div>
`;
